src/Haskell/Ide/Engine/Support/HieExtras.hs

```diff
@@ -194,16 +273,25 @@ safeTyThingId (AConLike (RealDataCon dc)) = Just $ dataConWrapId dc
 safeTyThingId _                           = Nothing
 
 -- Associates a module's qualifier with its members
-type QualCompls = Map.Map T.Text [CompItem]
+newtype QualCompls = QualCompls { getQualCompls :: Map.Map T.Text [CompItem] }
+
+instance Semigroup QualCompls where
+  (QualCompls a) <> (QualCompls b) = QualCompls $ Map.unionWith (++) a b
+
+instance Monoid QualCompls where
+  mempty = QualCompls Map.empty
 
 data CachedCompletions = CC
   { allModNamesAsNS :: [T.Text]
   , unqualCompls :: [CompItem]
   , qualCompls :: QualCompls
   , importableModules :: [T.Text]
-  , cachedExtensions :: [T.Text]
   } deriving (Typeable)
 
+-- The supported languages and extensions
+languagesAndExts :: [T.Text]
+languagesAndExts = map T.pack GHC.supportedLanguagesAndExtensions
```

Comment:
This is great, was this previously cached for a reason? Helpful reading by the way https://stackoverflow.com/a/3951092/1320911

Reply:
This basically achieve the same as caching it. When I added it to the cache I just wanted to avoid the text packing on each use.